OrbitCore/SymbolHelper.cpp

```diff
@@ -0,0 +1,166 @@
+  std::string name_without_extension = Path::StripExtension(module->m_Name);
+
+  std::vector<std::string> search_file_paths;
+  for (const auto& directory : search_directories) {
+    search_file_paths.emplace_back(directory + name_without_extension +
+                                   ".debug");
+    search_file_paths.emplace_back(directory + module->m_Name + ".debug");
+  }
+
+  LOG("Trying to find symbols for module %s", module->m_Name.c_str());
+  for (const auto& symbols_file_path : search_file_paths) {
+    // LOG("Looking at path %s", symbols_file_path.c_str());
+    if (!Path::FileExists(symbols_file_path)) continue;
+
+    std::unique_ptr<ElfFile> symbols_file = ElfFile::Create(symbols_file_path);
+    if (!symbols_file) continue;
+    if (!symbols_file->HasSymtab()) continue;
+    if (symbols_file->GetBuildId() != module->m_DebugSignature) continue;
+
+    return LoadFromElfFile(module, symbols_file);
+  }
+
+  return false;
+}
+
+bool SymbolHelper::LoadSymbolsIncludedInBinary(std::shared_ptr<Module> module) {
+  std::unique_ptr<ElfFile> elf_file = ElfFile::Create(module->m_FullName);
+
+  if (!elf_file) {
+    ERROR("Unable to load module as elf-file: %s", module->m_FullName.c_str());
+    return false;
+  }
+
+  if (!elf_file->HasSymtab()) return false;
+  return LoadFromElfFile(module, elf_file);
+}
+
+bool SymbolHelper::LoadSymbolsCollector(std::shared_ptr<Module> module) {
+  if (LoadSymbolsIncludedInBinary(module)) return true;
+
+  std::vector<std::string> search_directories =
+      SymbolHelper::GetCollectorSymbolLocations();
+  search_directories.emplace_back(Path::GetDirectory(module->m_FullName));
+
+  return FindAndLoadSymbols(module, search_directories);
+}
+
+bool SymbolHelper::LoadSymbolsVisualizer(std::shared_ptr<Module> module) {
+  return FindAndLoadSymbols(module,
+                            SymbolHelper::GetVisualizerSymbolLocations());
+}
+
+void SymbolHelper::LoadSymbolsFromDebugInfo(std::shared_ptr<Module> module,
+                                            ModuleDebugInfo& module_info) {
+  module->m_Pdb =
+      std::make_shared<Pdb>(module->m_AddressStart, module_info.load_bias,
+                            module_info.m_PdbName, module->m_FullName);
+
+  for (Function& function : module_info.m_Functions) {
+    module->m_Pdb->AddFunction(function);
+  }
+  module->m_Pdb->ProcessData();
+  module->m_PdbName = module_info.m_PdbName;
+  module->SetLoaded(true);
+}
+
+void SymbolHelper::FillDebugInfoFromModule(std::shared_ptr<Module> module,
+                                           ModuleDebugInfo& module_info) {
+  FAIL_IF(!module->m_Pdb, "Pdb not initiallized for module %s",
+          module->m_Name.c_str());
+
+  module_info.m_Name = module->m_Name;
+  module_info.m_Functions = module->m_Pdb->GetFunctions();
+  module_info.load_bias = module->m_Pdb->GetLoadBias();
+  module_info.m_PdbName = module->m_PdbName;
+}
+
+std::vector<std::string> SymbolHelper::GetVisualizerSymbolLocations() {
```

Comment:
Why Visualizer? Is it jus reading search paths from file?

Reply:
True, changed it